libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.kt

```diff
@@ -0,0 +1,1687 @@
+import com.salesforce.androidsdk.developer.support.notifications.local.ShowDeveloperSupportNotifier.Companion.showDeveloperSupportNotification
+import com.salesforce.androidsdk.push.PushMessaging.UNREGISTERED_ATTEMPT_COMPLETE_EVENT
+import com.salesforce.androidsdk.push.PushMessaging.isRegistered
+import com.salesforce.androidsdk.push.PushMessaging.unregister
+import com.salesforce.androidsdk.push.PushNotificationInterface
+import com.salesforce.androidsdk.push.PushService
+import com.salesforce.androidsdk.rest.ClientManager
+import com.salesforce.androidsdk.rest.ClientManager.LoginOptions
+import com.salesforce.androidsdk.rest.RestClient
+import com.salesforce.androidsdk.security.BiometricAuthenticationManager
+import com.salesforce.androidsdk.security.SalesforceKeyGenerator.getEncryptionKey
+import com.salesforce.androidsdk.security.ScreenLockManager
+import com.salesforce.androidsdk.ui.AccountSwitcherActivity
+import com.salesforce.androidsdk.ui.DevInfoActivity
+import com.salesforce.androidsdk.ui.LoginActivity
+import com.salesforce.androidsdk.util.AuthConfigUtil.getMyDomainAuthConfig
+import com.salesforce.androidsdk.util.EventsObservable
+import com.salesforce.androidsdk.util.EventsObservable.EventType.AppCreateComplete
+import com.salesforce.androidsdk.util.EventsObservable.EventType.LogoutComplete
+import com.salesforce.androidsdk.util.SalesforceSDKLogger.e
+import com.salesforce.androidsdk.util.SalesforceSDKLogger.w
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.Default
+import kotlinx.coroutines.Dispatchers.Main
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withTimeout
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import org.json.JSONObject
+import java.lang.String.CASE_INSENSITIVE_ORDER
+import java.net.URI
+import java.util.Locale.US
+import java.util.SortedSet
+import java.util.UUID.randomUUID
+import java.util.concurrent.ConcurrentSkipListSet
+import java.util.regex.Pattern
+import com.salesforce.androidsdk.auth.idp.IDPManager as DefaultIDPManager
+import com.salesforce.androidsdk.auth.idp.SPManager as DefaultSPManager
+import com.salesforce.androidsdk.security.interfaces.BiometricAuthenticationManager as DefaultBiometricAuthenticationManager
+
+/**
+ * This class serves as an interface to the various functions of the Salesforce
+ * SDK. In order to use the SDK, your app must first instantiate the singleton
+ * SalesforceSDKManager object by calling the static init() method. After
+ * calling init(), use the static getInstance() method to access the singleton
+ * SalesforceSDKManager object.
+ *
+ * TODO: Remove the Android `Context` property of this object, as it is a memory leak.
+ *
+ * @param context The Android context
+ * @param mainActivity Activity that should be launched after the login flow
+ * @param loginActivity Login activity
+ */
+open class SalesforceSDKManager protected constructor(
```

Comment:
Obviously, this is one of the most significant Kotlin conversions we've done.  I was concerned it would introduce more API change or regression than desired at this time.  In the end, though, it went really smoothly and I able to undo all of the API changes.  The only remaining API change we might see is inspector warnings regarding optionals and nullability.